images/sources/class-diagram.puml

```diff
@@ -0,0 +1,53 @@
+@startuml
+!theme plain
+skinparam linetype ortho
+
+together {
+    Class "DAPT Script" {
+        Script Type
+        Primary Language
+    }
+
+    Class "Script Event" {
+        Script Event Identifier
+        Begin
+        End
+        {field} (optional) Script Event Type
+        {field} (optional) Script Event Description
+        {field} (optional) On Screen
+    }
+
+    Class "Text" {
+        Text content
+        Text Language Source
+        {field} (optional) Language
+    }
+}
+
+together {
+    Class "Character" {
+        Character Identifier
+        Name
+        {field} (optional) Talent Name
+    }
+
+    Class "Style" {
+        Style Identifier
+        {field} (optional) Character Identifier
+        Style Attributes
+    }
+}
+
+"DAPT Script" "0" *-- "many" "Script Event" : contains
+"DAPT Script" "0" *-- "many" "Character" : contains
+"DAPT Script" "0" *-- "many" "Style" : contains
+"Character" "0" .. "many" "Style" : Character Style
+"Script Event" "0" .. "many" "Style" : Character Style
+"Text" "0" .. "many" "Style" : Character Style
+"Text" "0" .. "many" "Style" : Other Style
+"Script Event" "0" *-- "many" "Text" : contains
```

Comment:
Can we replace many by 2? A Script Event contains at most 2 Text objects.

Reply:
I don't think so, as things stand. For example a Script Event can have Original language, Pivot translation and Target language translation which would have to be 3 Text objects. If there's a need to constrain the contents per script type, that would be an additional constraint, but in general, it's 0 to many as we've written the spec at the moment.

Reply:
By the way, all the cardinalities made no sense and have now been fixed to have ranges on both ends of the arrow where appropriate.